### PR TITLE
Show pictographs for completion items and description (inspired by Te…

### PR DESCRIPTION
**Pull request by as3boyan**
…rn project from start)

### CODE REVIEW

**Comment by as3boyan** [comment]
Code logic is such, that if completion for something becomes not available - completion is still shown, it should be closed

I don't know if we had this issue before

How to reproduce:
activate top level completion
type "Str"
type "."
completion for top level is still shown

Should be fixed.

File: src/Completion.hx [hunk 1]
```diff
@@ -2,46 +2,58 @@ package;
...
+	var WORD:EReg = ~/[A-Z]+$/i;
+	var RANGE = 500;
+	public var curWord:String;
+	public var completions:Array<CompletionItem> = [];
+	var backupDocValue:String;
 	
-	public static function registerHelper() 
+	static var instance:Completion;
+	
+	public static function get()
```
Comment (clemos) [inline src/Completion.hx]: Not a big fan of singletons :|
How about making just using `new`, retaining the `Completion` instance in `Editor`, and pass it to `FunctionParameterHelper`, `ColorPreview`, etc  ?

Comment (clemos) [inline src/Completion.hx]: (`ColorPreview` should have a constructor too)

File: src/FunctionParametersHelper.hx [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	var lastPos:Pos;
+	
+	static var instance:FunctionParametersHelper = null;
+	
+	public function new()
+	{
+			
+	}
+	
+	public static function get()
```
Comment (clemos) [inline src/FunctionParametersHelper.hx]: same comment as for `Completion` : 
how about using the constructor, and keeping the instance in `Editor` ?

File: src/api/Compiler.hx [hunk 3]
```diff
@@ -179,12 +180,24 @@ class Compiler {
...
 			if (completionType == CompletionType.DEFAULT)
 			{
 				for( e in xml.nodes.i ){
-					var w = e.att.n;
+					var w:CompletionItem = {n: e.att.n, d: ""};
+					
+					if (e.hasNode.t)
+					{
+						w.t = e.node.t.innerData;
+						w.d = w.t + "</br>";
```
Comment (clemos) [inline src/api/Compiler.hx]: `<br/>` :S ?

**Comment by clemos** [comment]
BTW, I can totally do the changes some day, if you agree with them of course ;)

File: src/api/Compiler.hx (hunk 3 above)
Comment (as3boyan) [inline src/api/Compiler.hx]: Maybe, but this one seem to work too somehow.

File: src/Completion.hx (hunk 1 above)
Comment (as3boyan) [inline src/Completion.hx]: Ok, not a problem

File: src/api/Compiler.hx (hunk 3 above)
Comment (clemos) [inline src/api/Compiler.hx]: Sorry, that's not what I meant :D (I hadn't even figured it was weird)
Whether it is `<br>` `<br/>` or `</br>`, I still think it shouldn't be in the API for SoC's sake :\

**Comment by as3boyan** [comment]
Ok it should be fixed now

**Comment by clemos** [comment]
It's online, well done :)
I've made a few tweaks regarding those `<br/>`s, let me know if all is fine.

**Comment by as3boyan** [comment]
It is ok

**Pull request merged**